### PR TITLE
Begin new requester, implemented getDefaultConfig

### PR DESCRIPTION
**Pull request by etekweb**
- Set up base file structure for new requester and initializer
- Replicated getDefaultConfig functionality in new structure
- Copied over constants, errors, and (needed) types for new structure

### CODE REVIEW

File: src/extractors/youtube-new/utils/errors.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+abstract class PageError extends YoutubeError {
+  id: string;
+  constructor(id: string, message: string, details?: unknown) {
+    super(message, details);
+    // Object.setPrototypeOf(this, PageError.prototype);
+    this.id = id;
+  }
+}
+
+abstract class VideoError extends PageError { }
```
Comment (404-Program-not-found) [inline src/extractors/youtube-new/utils/errors.ts]: We don't need these error objects anymore. Just use the ones under `src/utils/errors.ts` with an error message instead

File: src/extractors/youtube-new/utils/constants.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    YT_URL: "https://www.youtube.com",
+    YT_MOBILE: "https://m.youtube.com",
+    YT_MUSIC_URL: "https://music.youtube.com",
+    YT_BASE_API: "https://www.youtube.com/youtubei/v1",
+    YT_SUGGESTION_API: "https://suggestqueries.google.com/complete/search",
+  },
+  YTAPIVAL: {
+    VERSION: "17.20",
+    CLIENTNAME: "ANDROID",
+    VERSION_WEB: "2.20220411.09.00",
```
Comment (404-Program-not-found) [inline src/extractors/youtube-new/utils/constants.ts]: Upgrade to Version 2.20230206.06.00 just so that we have the latest features when developing

**Review by 404-Program-not-found — CHANGES_REQUESTED** [review]
Just a few small changes, otherwise LGTM

File: src/extractors/youtube-new/utils/constants.ts (hunk 2 above)
Comment (etekweb) [reply src/extractors/youtube-new/utils/constants.ts]: updated in fd1001e

File: src/extractors/youtube-new/utils/errors.ts (hunk 1 above)
Comment (etekweb) [reply src/extractors/youtube-new/utils/errors.ts]: updated in e84daeb

**Review by 404-Program-not-found — APPROVED** [review]

**Pull request merged**